README.md

```diff
@@ -1,23 +1,94 @@
-# Onigumo #
+# Onigumo
 
-## About ##
+## O projektu
 
-This is an attempt to build just another web-crawler, also called _Spider_. Its purpose is to get data from a website in a form of a list of objects. This data can be then used to download linked files and place them in a database or a folder structure.
+Onigumo je jeden z dalších web-crawlerů, také známých pod pojmem _Spider_.
+Onigumo obecně řídí toky dat, určuje směry toků a v jaké chvíli se
+aktivuje příslušná operace.
+Konrétně slouží k procházení webových aplikací či statických webových stránek.
+Obsah a případně další příslušná metadata uloží do strukturované podoby,
+která je vhodná pro další strojové zpracování. K dosažení tohoto cíle je
+Onigumo rozděleno do tří vzájemně propojených logických celků
+(Operator, Downloader, Parser), které tvoří procesní workflow.
 
-## Architecture ##
+## Arhitektura
 
-### Building blocks ###
+Jádrem Oniguma jsou tři základní moduly:
+* Operator
+* Downloader
+* Parser
 
-The application uses Spiders (Ruby modules) containing workflow and data-mining methods to get data from a website. A Spider’s public interface consists of workflows how to get all the wanted data from the server. Typically get a page, parse it and possibly get more pages using the parsed data.
+jejichž vzájemná spolupráce je znázorněna na diagramu níže
 
-### Mechanism ###
+```mermaid
+flowchart LR
+    START             -->           operator(OPERATOR)
+    operator    -- urls.txt --->    downloader(DOWNLOADER)
+    downloader  -- *.html --->      parser(PARSER)
+    parser      -- *.json -->       operator
+```
 
-Scraping starts by inserting the first workflow action on a queue. This action doesn’t take any input an thus doesn’t have to wait for any data to be parsed. Any action can queue another action that uses the data parsed from a download page. Downloading and parsing ends when there is no more actions to be launched.
+### Operator
+Operator vytváří frontu url adres pro `Donwloader`. Tato fronta je vytvořena
+z počáteční uživatelské konfigurace a zároveň z výstupních dat modulu `Parser`
```

Comment:
I don’t feel like “z počáteční uživatelské konfigurace” is a good description. Although it may be technically correct, I don’t feel a hardcoded value to be a “configuration.”.